From 7633cd7fc972685c65f95cb42de693a0014f7815 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 1 Sep 2023 14:07:48 +0100
Subject: [PATCH 0549/1085] ASoC: bcm: Remove dependency on BCM2835 I2S

These soundcard drivers don't rely on a specific I2S interface, so
remove the dependency declarations.

See: https://github.com/raspberrypi/linux-2712/issues/111

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 sound/soc/bcm/Kconfig | 40 +---------------------------------------
 1 file changed, 1 insertion(+), 39 deletions(-)

--- a/sound/soc/bcm/Kconfig
+++ b/sound/soc/bcm/Kconfig
@@ -29,13 +29,11 @@ config SND_BCM63XX_I2S_WHISTLER
 
 config SND_BCM2708_SOC_CHIPDIP_DAC
          tristate "Support for the ChipDip DAC"
-         depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
          help
           Say Y or M if you want to add support for the ChipDip DAC soundcard
 
 config SND_BCM2708_SOC_GOOGLEVOICEHAT_SOUNDCARD
 	tristate "Support for Google voiceHAT soundcard"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_VOICEHAT
 	select SND_RPI_SIMPLE_SOUNDCARD
 	help
@@ -43,7 +41,6 @@ config SND_BCM2708_SOC_GOOGLEVOICEHAT_SO
 
 config SND_BCM2708_SOC_HIFIBERRY_DAC
         tristate "Support for HifiBerry DAC"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_PCM5102A
         select SND_RPI_SIMPLE_SOUNDCARD
         help
@@ -51,7 +48,6 @@ config SND_BCM2708_SOC_HIFIBERRY_DAC
 
 config SND_BCM2708_SOC_HIFIBERRY_DACPLUS
         tristate "Support for HifiBerry DAC+"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_PCM512x
         select SND_SOC_TPA6130A2
         select COMMON_CLK_HIFIBERRY_DACPRO
@@ -60,7 +56,6 @@ config SND_BCM2708_SOC_HIFIBERRY_DACPLUS
 
 config SND_BCM2708_SOC_HIFIBERRY_DACPLUSHD
         tristate "Support for HifiBerry DAC+ HD"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_PCM179X_I2C
         select COMMON_CLK_HIFIBERRY_DACPLUSHD
         help
@@ -68,7 +63,6 @@ config SND_BCM2708_SOC_HIFIBERRY_DACPLUS
 
 config SND_BCM2708_SOC_HIFIBERRY_DACPLUSADC
         tristate "Support for HifiBerry DAC+ADC"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_PCM512x_I2C
 	select SND_SOC_DMIC
         select COMMON_CLK_HIFIBERRY_DACPRO
@@ -77,7 +71,6 @@ config SND_BCM2708_SOC_HIFIBERRY_DACPLUS
 
 config SND_BCM2708_SOC_HIFIBERRY_DACPLUSADCPRO
         tristate "Support for HifiBerry DAC+ADC PRO"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_PCM512x_I2C
         select SND_SOC_PCM186X_I2C
         select SND_SOC_TPA6130A2
@@ -87,29 +80,25 @@ config SND_BCM2708_SOC_HIFIBERRY_DACPLUS
 
 config SND_BCM2708_SOC_HIFIBERRY_DACPLUSDSP
         tristate "Support for HifiBerry DAC+DSP"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_RPI_SIMPLE_SOUNDCARD
         help
          Say Y or M if you want to add support for HifiBerry DSP-DAC.
 
 config SND_BCM2708_SOC_HIFIBERRY_DIGI
         tristate "Support for HifiBerry Digi"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_WM8804
         help
          Say Y or M if you want to add support for HifiBerry Digi S/PDIF output board.
 
 config SND_BCM2708_SOC_HIFIBERRY_AMP
         tristate "Support for the HifiBerry Amp"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_TAS5713
         select SND_RPI_SIMPLE_SOUNDCARD
         help
          Say Y or M if you want to add support for the HifiBerry Amp amplifier board.
 
- config SND_BCM2708_SOC_PIFI_40
+config SND_BCM2708_SOC_PIFI_40
          tristate "Support for the PiFi-40 amp"
-         depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
          select SND_SOC_TAS571X
          select SND_PIFI_40
          help
@@ -117,7 +106,6 @@ config SND_BCM2708_SOC_HIFIBERRY_AMP
 
 config SND_BCM2708_SOC_RPI_CIRRUS
         tristate "Support for Cirrus Logic Audio Card"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_WM5102
         select SND_SOC_WM8804
         help
@@ -126,7 +114,6 @@ config SND_BCM2708_SOC_RPI_CIRRUS
 
 config SND_BCM2708_SOC_RPI_DAC
         tristate "Support for RPi-DAC"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_PCM1794A
         select SND_RPI_SIMPLE_SOUNDCARD
         help
@@ -134,14 +121,12 @@ config SND_BCM2708_SOC_RPI_DAC
 
 config SND_BCM2708_SOC_RPI_PROTO
 	tristate "Support for Rpi-PROTO"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_WM8731_I2C
 	help
 	  Say Y or M if you want to add support for Audio Codec Board PROTO (WM8731).
 
 config SND_BCM2708_SOC_JUSTBOOM_BOTH
 	tristate "Support for simultaneous JustBoom Digi and JustBoom DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_WM8804
 	select SND_SOC_PCM512x
 	help
@@ -153,14 +138,12 @@ config SND_BCM2708_SOC_JUSTBOOM_BOTH
 
 config SND_BCM2708_SOC_JUSTBOOM_DAC
 	tristate "Support for JustBoom DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM512x
 	help
 	  Say Y or M if you want to add support for JustBoom DAC.
 
 config SND_BCM2708_SOC_JUSTBOOM_DIGI
 	tristate "Support for JustBoom Digi"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_WM8804
         select SND_RPI_WM8804_SOUNDCARD
 	help
@@ -168,21 +151,18 @@ config SND_BCM2708_SOC_JUSTBOOM_DIGI
 
 config SND_BCM2708_SOC_IQAUDIO_CODEC
 	tristate "Support for IQaudIO-CODEC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_DA7213
 	help
 	  Say Y or M if you want to add support for IQaudIO-CODEC.
 
 config SND_BCM2708_SOC_IQAUDIO_DAC
 	tristate "Support for IQaudIO-DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM512x_I2C
 	help
 	  Say Y or M if you want to add support for IQaudIO-DAC.
 
 config SND_BCM2708_SOC_IQAUDIO_DIGI
 	tristate "Support for IQAudIO Digi"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_WM8804
 	select SND_RPI_WM8804_SOUNDCARD
 	help
@@ -190,14 +170,12 @@ config SND_BCM2708_SOC_IQAUDIO_DIGI
 
 config SND_BCM2708_SOC_I_SABRE_Q2M
         tristate "Support for Audiophonics I-Sabre Q2M DAC"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_I_SABRE_CODEC
         help
         Say Y or M if you want to add support for Audiophonics I-SABRE Q2M DAC
 
 config SND_BCM2708_SOC_ADAU1977_ADC
 	tristate "Support for ADAU1977 ADC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_ADAU1977_I2C
 	select SND_RPI_SIMPLE_SOUNDCARD
 	help
@@ -205,35 +183,30 @@ config SND_BCM2708_SOC_ADAU1977_ADC
 
 config SND_AUDIOINJECTOR_PI_SOUNDCARD
 	tristate "Support for audioinjector.net Pi add on soundcard"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_WM8731_I2C
 	help
 	  Say Y or M if you want to add support for audioinjector.net Pi Hat
 
 config SND_AUDIOINJECTOR_OCTO_SOUNDCARD
 	tristate "Support for audioinjector.net Octo channel (Hat) soundcard"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_CS42XX8_I2C
 	help
 	  Say Y or M if you want to add support for audioinjector.net octo add on
 
 config SND_AUDIOINJECTOR_ISOLATED_SOUNDCARD
 	tristate "Support for audioinjector.net isolated DAC and ADC soundcard"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_CS4271_I2C
 	help
 	  Say Y or M if you want to add support for audioinjector.net isolated soundcard
 
 config SND_AUDIOSENSE_PI
 	tristate "Support for AudioSense Add-On Soundcard"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_TLV320AIC32X4_I2C
 	help
 	  Say Y or M if you want to add support for tlv320aic32x4 add-on
 
 config SND_DIGIDAC1_SOUNDCARD
         tristate "Support for Red Rocks Audio DigiDAC1"
-        depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
         select SND_SOC_WM8804
         select SND_SOC_WM8741
         help
@@ -241,35 +214,30 @@ config SND_DIGIDAC1_SOUNDCARD
 
 config SND_BCM2708_SOC_DIONAUDIO_LOCO
 	tristate "Support for Dion Audio LOCO DAC-AMP"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM5102a
 	help
 	  Say Y or M if you want to add support for Dion Audio LOCO.
 
 config SND_BCM2708_SOC_DIONAUDIO_LOCO_V2
 	tristate "Support for Dion Audio LOCO-V2 DAC-AMP"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM5122
 	help
 	  Say Y or M if you want to add support for Dion Audio LOCO-V2.
 
 config SND_BCM2708_SOC_ALLO_PIANO_DAC
 	tristate "Support for Allo Piano DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM512x_I2C
 	help
 	  Say Y or M if you want to add support for Allo Piano DAC.
 
 config SND_BCM2708_SOC_ALLO_PIANO_DAC_PLUS
 	tristate "Support for Allo Piano DAC Plus"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM512x_I2C
 	help
 	  Say Y or M if you want to add support for Allo Piano DAC Plus.
 
 config SND_BCM2708_SOC_ALLO_BOSS_DAC
 	tristate "Support for Allo Boss DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_PCM512x_I2C
 	select COMMON_CLK_HIFIBERRY_DACPRO
 	help
@@ -277,7 +245,6 @@ config SND_BCM2708_SOC_ALLO_BOSS_DAC
 
 config SND_BCM2708_SOC_ALLO_BOSS2_DAC
 	tristate "Support for Allo Boss2 DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	depends on I2C
 	select REGMAP_I2C
 	select SND_AUDIO_GRAPH_CARD
@@ -286,7 +253,6 @@ config SND_BCM2708_SOC_ALLO_BOSS2_DAC
 
 config SND_BCM2708_SOC_ALLO_DIGIONE
 	tristate "Support for Allo DigiOne"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_WM8804
 	select SND_RPI_WM8804_SOUNDCARD
 	help
@@ -294,7 +260,6 @@ config SND_BCM2708_SOC_ALLO_DIGIONE
 
 config SND_BCM2708_SOC_ALLO_KATANA_DAC
 	tristate "Support for Allo Katana DAC"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	depends on I2C
 	select REGMAP_I2C
 	select SND_AUDIO_GRAPH_CARD
@@ -303,14 +268,12 @@ config SND_BCM2708_SOC_ALLO_KATANA_DAC
 
 config SND_BCM2708_SOC_FE_PI_AUDIO
 	tristate "Support for Fe-Pi-Audio"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_SGTL5000
 	help
 	  Say Y or M if you want to add support for Fe-Pi-Audio.
 
 config SND_PISOUND
 	tristate "Support for Blokas Labs pisound"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_RAWMIDI
 	help
 	  Say Y or M if you want to add support for Blokas Labs pisound.
@@ -328,7 +291,6 @@ config SND_RPI_WM8804_SOUNDCARD
 
 config SND_DACBERRY400
 	tristate "Support for DACBERRY400 Soundcard"
-	depends on SND_BCM2708_SOC_I2S || SND_BCM2835_SOC_I2S
 	select SND_SOC_TLV320AIC3X_I2C
 	help
 	  Say Y or M if you want to add support for tlv320aic3x add-on
